pkg/operator/controller/status_test.go

```diff
@@ -71,14 +151,18 @@ func TestComputeOperatorStatusConditions(t *testing.T) {
 		} else {
 			available = configv1.ConditionFalse
 		}
+		if tc.outputs.transition {
+			transitionTime = metav1.Now()
+		}
 		expected := []configv1.ClusterOperatorStatusCondition{
 			{
 				Type:   configv1.OperatorFailing,
 				Status: failing,
 			},
 			{
-				Type:   configv1.OperatorProgressing,
-				Status: progressing,
+				Type:               configv1.OperatorProgressing,
+				Status:             progressing,
+				LastTransitionTime: transitionTime,
```

Comment:
We don't look at `LastTransitionTime` when computing `gotExpected` below.  I think we need to add something like the following to the condition comparison loop below:
```go
					if conditionA.Type == configv1.OperatorProgressing && tc.outputs.transition && conditionA.LastTransitionTime.IsZero() {
						t.Errorf("%q: expected transition time to be updated", tc.description)
					}
```

Reply:
Thanks for catching this. I'll update the test.

Reply:
This might be trickier than I originally thought... If the condition does not exist in the conditions slice that the test passes to `computeOperatorStatusConditions`, then `computeOperatorStatusConditions` will _always_ set it, so it will _always_ have have a non-zero `LastTransitionTime`.  So we need to have at least a `Progressing` condition already set in the input to `computeOperatorStatusConditions`.  I'm starting to think it would be worth adding a separate test function for testing that `Progressing` gets updated on upgrades.  What do you think?

Reply:
I'm in the process of refactoring the test. Since it's expected that updating a condition message triggers `LastTransitionTime`, I will create a separate test that checks the output message when versions are equal and not equal.